FileProcessor/src/main/java/com/group7/FileProcessor/Sequencer.java

```diff
@@ -1,136 +1,111 @@
-        }
-
-    }
-
-}
+/*
+ * Lukas Krampitz and Group 7
+ * Mar 29, 2023
+ * The main sequencer for the file processor. This class is the main high level logic control of the file processor
+ */
+package com.group7.FileProcessor;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.group7.FileProcessor.pojo.EntriesPOJO;
+import com.group7.FileProcessor.pojo.ParametersPOJO;
+import com.group7.FileProcessor.pojo.ProcessingElementPOJO;
+import com.group7.FileProcessor.pojo.ScenarioPOJO;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ *
+ * @author Tacitor
+ */
+public class Sequencer {
+
+    //Attributes
+    private ArrayList<ProcessingElementPOJO> processingElements; //Type of ProcessingElementPOJO is temporary. Will need the ProcessingElement interface when it is ready
+
+    private boolean readyToProcess; //Control and protection var. If this is true then the sequencer may run. If this is false then there has not been a scenarioAsString stored, or the previous scenario is done can a new one needs to be selected.
+    private String scenarioAsString; //The .JSON file that contains the senario as a String. Line for line. Passed in by the Main.java class.
+
+    /**
+     * Primary constructor for the Sequencer
+     */
+    public Sequencer() {
+        readyToProcess = false;
+        scenarioAsString = null; //Defaults to null when first run.
+    }
+
+    /**
+     * Accessor for the scenario as a String.
+     *
+     * @return - the scenario string
+     */
+    public String getScenarioAsString() {
+        return scenarioAsString;
+    }
+
+    /**
+     * Mutator for the scenario as a String. Checks to ensure this is not a null
+     * string or an empty string
+     *
+     * @param senarioAsString
+     */
+    public synchronized void setScenarioAsString(String senarioAsString) {
+        //check for a null or empty string
+        if (senarioAsString != null && !senarioAsString.equalsIgnoreCase("")) {
+            this.scenarioAsString = senarioAsString;
+            //allow for processing
+            readyToProcess = true;
+        }
+    }
+
+    /**
+     * Begin running through the scenario. This requires the scenarioAsString to
+     * be provided
+     */
+    public synchronized void run() {
+
+        //check if the senario may run
+        if (readyToProcess) {
+
+            //Use Jackson to pasrse the scenario
+            //Code coppied from Scenario.java that Eric @Fluff-E wrote
+            ScenarioPOJO scenario = new ScenarioPOJO();
+            try {
+                JsonNode node = Json.parse(scenarioAsString);//make JsonNode
+                scenario = Json.fromJson(node, ScenarioPOJO.class); //map to class
+            } catch (IOException e) {
+                System.out.println("ERROR: IOException when mapping .Json file to ScenarioPOJO in Sequencer.java\n" + e);
+            }
+            scenario.print();
```

Comment:
idk if it's a good idea to take this line of code and all the following code out of the try-catch. Because if an exception were to occur all this code would still run even is the .JSON file was not correctly read. From a exception management perspective I think this should stay in the try-catch.

Reply:
I agree with you, that it makes sense to have  a try catch for the following code.  I left it commented out.  We can sandwich the whole thing again if you want, but it makes sense for the mapping to have it's own catch to identify that exact dependency.

Reply:
Oh I see what you mean yeah, that's a good idea. I will try and add that my my new dev branch. I think I can figure something out.